src/frontend/src/components/FilePicker/index.tsx

```diff
@@ -61,9 +61,9 @@ export default function FilePicker({
         sdsType="primary"
         sdsStyle="square"
         onClick={shouldConfirm ? openDialog : handleClick}
-        disabled={isDisabled}
+        disabled={isLoading}
       >
-        {text}
+        {isLoading ? "Loading..." : text}
```

Comment:
We previously had this in one of the parent components but not the rest.  Now if this shared component receives the optional "isLoading" flag, the text will change.  If isLoading is false or undefined the original text will show.